src/boot/efi/boot.c

```diff
@@ -1460,8 +1460,30 @@ static BOOLEAN config_entry_add_loader_auto(Config *config, EFI_HANDLE *device,
                 return FALSE;
 
         /* do not add an entry for ourselves */
-        if (loaded_image_path && StriCmp(loader, loaded_image_path) == 0)
-                return FALSE;
+        if (loaded_image_path) {
+                if (StriCmp(loader, loaded_image_path) == 0)
+                        return FALSE;
+                else {
+                        UINTN len = 100*1024;
```

Comment:
I'm a bit confused here. Shouldn't this `else` block be independent of the check for `loaded_image_path`?